Acrolinx.Sidebar/ISidebar.cs

```diff
@@ -187,12 +189,67 @@ public bool SupportCheckSelection
 
     public class CheckedEventArgs : CheckEventArgs
     {
-        internal CheckedEventArgs(string checkId, Range range) : base(checkId)
+        internal CheckedEventArgs(string checkId, Range range, JArray embedCheckInformation, string inputFormat) : base(checkId)
         {
             Range = range;
+            EmbedCheckInformation = new JObject();
+            EmbedCheckInformation["embedCheckInformation"] = embedCheckInformation;
+            if (inputFormat.ToUpper().Equals("XML"))
+                InputFormat = Format.XML;
+            else if (inputFormat.ToUpper().Equals("HTML"))
+                InputFormat = Format.HTML;
+            else if (inputFormat.ToUpper().Equals("MARKDOWN"))
+                InputFormat = Format.Markdown;
+            else if (inputFormat.ToUpper().Equals("TEXT"))
+                InputFormat = Format.Text;
+            else if (inputFormat.ToUpper().Equals("WORD_XML"))
+                InputFormat = Format.Word_XML;
+            else InputFormat = Format.Auto;
         }
 
         public Range Range { get; private set; }
+
+        public JObject EmbedCheckInformation { get; private set; }
+
+        public Format InputFormat { get; private set; }
+
+        public string GetEmbedCheckDataAsEmbeddableString()
+        {
+            XmlDocument doc = new XmlDocument();
+            string returnString = "";
+            if (InputFormat == Format.XML)
+            {
+                returnString = "<?acrolinxCheckData ";
+                foreach (JObject pair in EmbedCheckInformation["embedCheckInformation"])
+                {
+                    returnString = returnString + pair["key"] + "=\"" + pair["value"] + "\" ";
```

Comment:
It's not save if you build XML by hand... what if key starts with a number? What if value contains chars that are not allowed in XML and must be encoded?